### ISSUE TITLE
Getting error Check failed: (best_split_info.left_count) > (0) at /__w/1/s/python-package/compile/src/treelearner/serial_tree_learner.cpp, 

### ISSUE THREAD

**Issue by AnjaliSetiya** [microsoft/LightGBM#5179]
## Environment info

LightGBM version or commit hash:

Command(s) you used to install LightGBM

```shell

```I am getting the error on training data in LGBM. I am not able to find in teh documentation why this is happening. For earlier data this was not happening. Please let me know ho can this be resolved.

<!-- Put any additional environment information here -->


## Additional Comments
<!-- What else should we know? -->
****

**Comment by jameslamb** [comment]
Thanks for using LightGBM!

We're happy to help, but need you to provide more information.

Unfortunately with only an error message, it's not possible for us to help you without a very significant investment of effort and guesswork. For example, you haven't told us whether you are using Python, R, Java, C, etc.

Please provide the information that was asked for, like a reproducible example (code that demonstrates the problem), version of LightGBM you are using, how you installed it, etc.

**Comment by jameslamb** [comment]
I think you may also find https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax helpful for formatting text on GitHub.

**Comment by samFarrellDay** [comment]
@AnjaliSetiya I get this error when I have set `min_sum_hessian_in_leaf` == 0. To fix it, I set `min_sum_hessian_in_leaf` equal to some very small value, like 0.0000001. This might work in your case. @jameslamb there are several open issues referencing this error. #4739, #4946, #4286, #3679

**Comment by jameslamb** [comment]
> This bug

Thanks for linking those!

But to be clear, an error message is not the same as a "bug". Without more information from @AnjaliSetiya , it's not possible to determine whether the issue they're facing shares the same root cause as those others you've linked.

**Comment by AnjaliSetiya** [comment]
In my case I'm using the library in Python. I have dataset of around 10000 rows and 12000 columns. I'm trying to standardize the model. I did the hyperpatameter tuning earlier on a particular dataset. With those range of hyperpatameters that I have, I want to compare accuracy of a slightly different data, let's call it data2. For that purpose I was using the model with the same hyperpatameter range for data 2 is when I got this error. I didn't get the error on intial data. Min_data_in_hessian_leaf has been set to range 0 to 0.0001.
Let me know if you have any solution.

**Comment by jameslamb** [comment]
Thanks for that! But that is still not enough information for us to reproduce this issue, or to suggest anything you can try to avoid it.

If you aren't able to provide the information I asked for in https://github.com/microsoft/LightGBM/issues/5179#issuecomment-1109724145, especially a reproducible example, then it isn't possible for us to tell whether whatever you're seeing is a different issue than others linked in https://github.com/microsoft/LightGBM/issues/5179#issuecomment-1109749525, and we'll close this issue.

**Comment by AnjaliSetiya** [comment]
In my case I'm using the library in Python. I have dataset of around 10000
rows and 12000 columns. I'm trying to standardize the model. I did the
hyperpatameter tuning earlier on a particular dataset. With those range of
hyperpatameters that I have, I want to compare accuracy of a slightly
different data, let's call it data2. For that purpose I was using the model
with the same hyperpatameter range for data 2 is when I got this error. I
didn't get the error on intial data. Min_data_in_hessian_leaf has been set
to range 0 to 0.0001.
Let me know if you have any solution.
